vault/resource_kubernetes_secret_backend_role.go

```diff
@@ -63,8 +64,19 @@ func kubernetesSecretBackendRoleResource() *schema.Resource {
 					Type: schema.TypeString,
 				},
 				Description: "The list of Kubernetes namespaces this role can generate " +
-					"credentials for. If set to '*' all namespaces are allowed.",
-				Required: true,
+					"credentials for. If set to '*' all namespaces are allowed. If set with" +
+					"`allowed_kubernetes_namespace_selector`, the conditions are `OR`ed.",
+				Optional: true,
+			},
+			fieldAllowedKubernetesNamespaceSelector: {
+				Type: schema.TypeString,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
```

Comment:
What is the purpose of `Elem`? Why not just `Type: schema.TypeString`?

Reply:
Copy/paste mistake. Fixed.